### PR TITLE
Add Xaml command hanler for CreateEventHandler command

### PR DESCRIPTION
**Pull request by LinglingTong**
This PR added code to handler when the CreateEventHandler command gets triggered and allow XAML language service to execute the command. 

When a XamlCompletionItem has an EventDescription, we will set this CreateEventHandlerCommand on the CompletionItem, and the command will be triggered when the completionItem gets committed.

### CODE REVIEW

File: src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Commands/CreateEventCommandHandlerProvider.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+using Microsoft.CodeAnalysis.Editor.Xaml;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.CodeAnalysis.LanguageServer.Handler.Commands;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.VisualStudio.LanguageServices.Xaml.Implementation.LanguageServer.Handler.Commands
+{
+    [ExportLspRequestHandlerProvider(StringConstants.XamlLanguageName), Shared]
+    [ProvidesMethod(Methods.WorkspaceExecuteCommandName)]
```
Comment (dibarbet) [inline src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Commands/CreateEventCommandHandlerProvider.cs:17]: I think you just want ProvidesCommand, ProvidesMethod shouldn't be needed

**Comment by LinglingTong** [comment]
@dibarbet @mgoertz-msft

File: src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Completion/CompletionHandler.cs [hunk 2]
```diff
@@ -89,6 +89,16 @@ private static CompletionItem CreateCompletionItem(XamlCompletionItem xamlComple
                 };
             }
 
+            if (xamlCompletion.EventDescription.HasValue)
+            {
+                item.Command = new Command()
+                {
+                    CommandIdentifier = StringConstants.CreateEventHandlerCommand,
+                    Arguments = new object[] { textDocument, xamlCompletion.EventDescription },
+                    Title = Resources.CreateEventHandler
```
Comment (dibarbet) [inline src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Completion/CompletionHandler.cs:98]: Is the title shown anywhere for completion commands?  If it's not we may not need to localize it.

File: src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Commands/CreateEventCommandHandler.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            var commandService = document.Project.LanguageServices.GetService<IXamlCommandService>();
+            if (commandService == null)
+            {
+                return false;
+            }
+
+            // request.Arguments has two argument for CreateEventHandlerCommand
+            // Arguments[0]: TextDocumentIdentifier
+            // Arguments[1]: XamlEventDescription
+            var arguments = new object[] { ((JToken)request.Arguments[1]).ToObject<XamlEventDescription>() };
```
Comment (dibarbet) [inline src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Commands/CreateEventCommandHandler.cs:51]: I don't see arguments[0] being used anywhere, do we actually need that item?

Comment (LinglingTong) [reply src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Commands/CreateEventCommandHandler.cs:51]: Yeah it is used in line 29 GetTextDocumentIdentifier.

File: src/VisualStudio/Xaml/Impl/Features/Commands/IXamlCommandService.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.VisualStudio.LanguageServices.Xaml.Features.Commands
+{
+    internal interface IXamlCommandService : ILanguageService
+    {
+        /// <summary>
+        /// Execute the <paramref name="command"/> with the <paramref name="commandArguments"/> on server
```
Comment (mgoertz-msft) [inline src/VisualStudio/Xaml/Impl/Features/Commands/IXamlCommandService.cs:15]: >on server [](start = 96, length = 9)

That's just an implementation detail that's not relevant to the description of the interface

File: src/VisualStudio/Xaml/Impl/Features/Commands/XamlEventDescription.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.VisualStudio.LanguageServices.Xaml.Features.Commands
+{
```
Comment (mgoertz-msft) [inline src/VisualStudio/Xaml/Impl/Features/Commands/XamlEventDescription.cs:8]: Feels like this should live under Completion and not Commands. We just happen to pass it around as a command parameter.

File: src/VisualStudio/Xaml/Impl/Features/Commands/IXamlCommandService.cs (hunk 4 above)
Comment (LinglingTong) [reply src/VisualStudio/Xaml/Impl/Features/Commands/IXamlCommandService.cs:15]: Sure I will update the comment.

File: src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Commands/CreateEventCommandHandlerProvider.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    [ProvidesCommand(StringConstants.CreateEventHandlerCommand)]
+    internal class CreateEventCommandHandlerProvider : AbstractRequestHandlerProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CreateEventCommandHandlerProvider()
+        {
+        }
+
```
Comment (mgoertz-msft) [inline src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Commands/CreateEventCommandHandlerProvider.cs:26]: Is this actually needed? We're not importing anything in here.

File: src/VisualStudio/Xaml/Impl/Features/Commands/XamlEventDescription.cs (hunk 5 above)
Comment (LinglingTong) [reply src/VisualStudio/Xaml/Impl/Features/Commands/XamlEventDescription.cs:8]: Yeah I have been moving this struct between Commands and Completion too. Now as you vote for completion I guess completion wins. :)

File: src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Commands/CreateEventCommandHandlerProvider.cs (hunk 6 above)
Comment (LinglingTong) [reply src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Commands/CreateEventCommandHandlerProvider.cs:26]: This is required by the ExportLspRequestHandlerProvider. I will get warning if I don't have an importing constructor.

File: src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Commands/CreateEventCommandHandler.cs (hunk 3 above)
Comment (dibarbet) [reply src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Commands/CreateEventCommandHandler.cs:51]: Ah I missed that.  It looks like if we get the command it must have that argument, so I'd recommend First()

File: src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Commands/CreateEventCommandHandlerProvider.cs (hunk 1 above)
Comment (LinglingTong) [reply src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Commands/CreateEventCommandHandlerProvider.cs]: Good catch! I have removed the ProvidesMethod.

File: src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Completion/CompletionHandler.cs (hunk 2 above)
Comment (LinglingTong) [reply src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Completion/CompletionHandler.cs]: You have a good point. I am not aware the title is shown anywhere so far. I have removed the string from the resources.

File: src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Commands/CreateEventCommandHandler.cs (hunk 3 above)
Comment (LinglingTong) [reply src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Commands/CreateEventCommandHandler.cs:52]: Sounds good to me.

File: src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Completion/CompletionHandler.cs (hunk 2 above)
Comment (mgoertz-msft) [reply src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Completion/CompletionHandler.cs]: Small improvement, make it a const string in this class.

---
In reply to: [587847403](https://github.com/dotnet/roslyn/pull/51670#discussion_r587847403) [](ancestors = 587847403)

**Review by mgoertz-msft — APPROVED** [review]
:shipit:

**Review by dibarbet — APPROVED** [review]

**Pull request merged**